platform/linux-generic/odp_packet.c

```diff
@@ -1910,6 +1910,92 @@ int _odp_packet_copy_md_to_packet(odp_packet_t srcpkt, odp_packet_t dstpkt)
 	return dst_uarea_size < src_uarea_size;
 }
 
+/* Simple implementation of ones complement sum.
+ * Based on RFC1071 and its errata.
+ */
+typedef union {
+	uint16_t w;
+	uint8_t b[2];
+} swap_buf_t;
+
+static uint32_t segment_sum16_32(const uint8_t *p,
+				 uint32_t len,
+				 uint32_t offset)
+
+{
+	uint32_t sum = 0;
+
+	/* Include second part of 16-bit short word split between segments */
+	if (len > 0 && (offset % 2)) {
+		swap_buf_t sw;
+
+		sw.b[0] = 0;
+		sw.b[1] = *(p++);
```

Comment:
Not wrong, but unnecessary as `*p++` is equivalent. See [here](https://www.geeksforgeeks.org/difference-between-p-p-and-p/) for a discussion. Same comment for other instances of this idiom.

Reply:
Fixed now